kms/src/main/java/kms/DecryptAsymmetric.java

```diff
@@ -56,11 +61,41 @@ public void decryptAsymmetric(
       CryptoKeyVersionName keyVersionName =
           CryptoKeyVersionName.of(projectId, locationId, keyRingId, keyId, keyVersionId);
 
+      // Optional, but recommended: compute ciphertext's CRC32C. See helpers below.
+      long ceiphertextCrc32c = getCrc32cAsLong(ciphertext);
+
       // Decrypt the ciphertext.
-      AsymmetricDecryptResponse response =
-          client.asymmetricDecrypt(keyVersionName, ByteString.copyFrom(ciphertext));
+      AsymmetricDecryptRequest request =
+          AsymmetricDecryptRequest.newBuilder()
+              .setName(keyVersionName.toString())
+              .setCiphertext(ByteString.copyFrom(ciphertext))
+              .setCiphertextCrc32C(
+                  Int64Value.newBuilder().setValue(ceiphertextCrc32c).build())
+              .build();
+      AsymmetricDecryptResponse response = client.asymmetricDecrypt(request);
+
+      // Optional, but recommended: perform integrity verification on response.
+      // For more details on ensuring E2E in-transit integrity to and from Cloud KMS visit:
+      // https://cloud.google.com/kms/docs/data-integrity-guidelines
+      if (!response.getVerifiedCiphertextCrc32C()) {
+        throw new IOException("AsymmetricDecrypt: request to server corrupted");
+      }
+
+      if (!crcMatches(response.getPlaintextCrc32C().getValue(),
+          response.getPlaintext().toByteArray())) {
+        throw new IOException("AsymmetricDecrypt: response from server corrupted");
+      }
+
       System.out.printf("Plaintext: %s%n", response.getPlaintext().toStringUtf8());
     }
   }
+
+  private long getCrc32cAsLong(byte[] data) {
+    return (long) Hashing.crc32c().hashBytes(data).asInt();
```

Comment:
Any reason you're using `asInt()` here? I tried this with `padToLong()' and the tests started to consistently pass. https://guava.dev/releases/19.0/api/docs/com/google/common/hash/HashCode.html#padToLong()